samples/boards/olimex_stm32_e407/ccm/README.rst

```diff
@@ -0,0 +1,37 @@
+.. _olimex_stm32_e407_ccm:
+
+Olimex STM32-E407 CCM example
+#############################
+
+Overview
+********
+
+Show usage of the Core Coupled Memory (CCM) that is available
+on several STM32 devices. The very important difference with
+normal RAM is that CCM can not be used for DMA.
```

Comment:
Maybe add how you can find out how much CCM is available? Are all allocations static?  I presume if you use too much you'l get a linker(?) error. 

Perhaps it would be good to add an introduction to the use of of CCM in the board documentation (e.g., https://www.zephyrproject.org/doc/boards/arm/olimex_stm32_e407/doc/olimex_stm32_e407.html) where you can document using the "__ccm_data_section, __ccm_bss_section, or __ccm_noinit_section" prefixes for using the CCM.

Reply:
"static" as in function-static and file-static (global)  variables, yes.  They just end up in a difference address range (normal RAM is under 0x2000_0000 and CCM under 0x1000_0000).  When you allocate the stack in CCM, you can also use it for stack local variables. But than you can not use stack variables for DMA, which will cause very obscure bugs, that are probably very hard to find. Also AFAIK you can not execute code from CCM, since it is only connected to the databus, but I have found no good docu about that. 

When using up more than there is the linker should complain.  I say "should" because I didn't test it yet (which I will now :-) ) 

Since you are Mr. Documentation, maybe you know a better place for the CCM docu. The CCM is not board specific, it is available on most STM32F4 and STM32F3 cpus. So is there be a place where I can document the CCM in general and than add a reference "This board supports CCM" in the board docu? A general "memory" section (maybe other CPU's have something similar?)  in the docu would be a nice place to really explain the ins and out, like getting a linker error when using to much, that DMA can not work from it, etc.

Reply:
Yes, It's very generic CCM since is a ST Specific modern way to implement the ARM "Tighly Coupled Memory" that existed on the ARMv6 based Cores like the Cortex-M1.

Reply:
@superna9999  OK I didn't know that, I thought it was a ST32 specific thing.  Any suggestions on how to improve this patch?